### PR TITLE
Ccn 185 view all catalogs amr - delete after merging

### PR DESCRIPTION
**Pull request by amrarick26**
Unit test for catalogs
Remove unnecessary resolve from catalogs state
Add type="submit" to Save Catalog button
Remove "Active" column from catalogs list view
Remove "Active" checkbox in catalog create view

### CODE REVIEW

File: catalogs/tests/catalogs.spec.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            Name: 'CatalogName'
+        }
+    }));
+
+    describe('State: catalogs', function(){
+        var state;
+        beforeEach(inject(function($state, OrderCloudParameters){
+            state = $state.get('catalogs', {}, {reload: true});
+            spyOn(OrderCloudParameters, 'Get').and.returnValue(null);
+            spyOn(oc.Catalogs, 'List').and.returnValue(null);
```
Comment (crhistianramirez) [inline catalogs/tests/catalogs.spec.js]: remove .and.returnValue(null) for both of these, its not necessary you can just do spyOn(OrderCloudParameters, 'Get')

Comment (crhistianramirez) [reply catalogs/tests/catalogs.spec.js]: We basically only want to return a value if we have to

File: catalogs/tests/catalogs.spec.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }));
+        it('should resolve Parameters', inject(function($injector, OrderCloudParameters){
+            $injector.invoke(state.resolve.Parameters);
+            expect(OrderCloudParameters.Get).toHaveBeenCalled();
+        }));
+        it('should resolve CatalogsList', inject(function($injector){
+            $injector.invoke(state.resolve.CatalogsList);
+            expect(oc.Catalogs.List).toHaveBeenCalled();
```
Comment (crhistianramirez) [inline catalogs/tests/catalogs.spec.js]: We always want to use toHaveBeenCalledWith over, just toHaveBeenCalled. So if you can think of a way to mock some parameters and expect them, then I would do that. For example on CatalogsList you can mock Parameters. One good thing to check is that if you don't set a pageSize it should still be set to 12

File: catalogs/tests/catalogs.spec.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }))
+    });
+
+    describe('Controller: CatalogCreateCtrl', function(){
+        var catalogCreateCtrl;
+        beforeEach(inject(function($state, $controller){
+            catalogCreateCtrl = $controller('CatalogCreateCtrl', {
+                $scope: scope
+            });
+            spyOn($state, 'go').and.returnValue(true);
```
Comment (crhistianramirez) [inline catalogs/tests/catalogs.spec.js]: Don't need to return anything here either.

File: catalogs/tests/catalogs.spec.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                defer.resolve(catalog);
+                spyOn(oc.Catalogs, 'Create').and.returnValue(defer.promise);
+                catalogCreateCtrl.saveCatalog();
+                scope.$digest();
+            });
+            it('should call the Catalogs Create method', function(){
+                expect(oc.Catalogs.Create).toHaveBeenCalledWith(catalog);
+            });
+            it('should enter the catalogs state and reload the state', inject(function($state){
+                expect($state.go).toHaveBeenCalledWith('catalogs', {}, {reload: true});
```
Comment (crhistianramirez) [inline catalogs/tests/catalogs.spec.js]: You can also test that toastr.success was called with 'Catalog Created', 'Success'

File: catalogs/templates/catalogs.tpl.html [hunk 5]
```diff
@@ -79,16 +76,13 @@
...
                 <td class="actions-cell">
                     <button class="btn btn-default md" ui-sref=".details({catalogid:catalog.ID})">
                         Details
                     </button>
-                    <button class="btn btn-default md" ui-sref=".createAssignment({catalogid:catalog.ID})">
-                        Assign
-                    </button>
+                    <!--<button class="btn btn-default md" ui-sref=".createAssignment({catalogid:catalog.ID})">-->
+                        <!--Assign-->
+                    <!--</button>-->
```
Comment (crhistianramirez) [inline catalogs/templates/catalogs.tpl.html]: If this doesn't need to be here, can we just delete it?

**Review by crhistianramirez — COMMENTED** [review]
Overall looks good. In addition to the comments I made it also looks like lines 101 - 113 can just be deleted from catalog.js if they aren't being used. 

Also, I didn't see any tests for CatalogsController. I know a lot of that filter/search functionality is being tested elsewhere but it is very possible that in the future something in that file gets changed and we won't know its wrong unless we have tests for it. Anything that can be tested, should.

**Review by robertsoniv — APPROVED** [review]

**Pull request merged**